### DISCUSSION TITLE
Smartmeter eBZ DD3 doesn't publish all values via mqtt since entering PIN

### DISCUSSION THREAD

**Discussion by tis-cs** [arendst/Tasmota#16337] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

After compiling a version with enabled scripts, I added and enabled the script for eBZ DD3 (OBIS) smart meter, which worked perfect. I got the following values all five seconds and pushed them to my mqtt broker. 

"SM Energie Bezung" --> object 1_8_0
"SM Energie Bezung T1" object 1_8_1
"SM Energie Bezung T2" object 1_8_2
"SM Identifikation"

The rest of all values were 0.000.

Now I got the pin of the smart meter to unlock the other values, which worked perfect.

Now i get all values in the webinterface but only the new values will be published to the mqtt broker every second:

"SM Leistung" --> 16_7_0
"SM Leistung L1" --> 36_7_0
"SM Leistung L2" --> 56_7_0
"SM Leistung L3" --> 76_7_0

My Problem is that the other values (1_8_X, smart meter counter) won't be published anymore via mqtt.
What can I do or what do I have to change in the script?

SCRIPT
`
>D
>B
TelePeriod 30
=>sensor53 r
>M 1
; Device: eBZ DD3 2R06 ODZ1
; protocol is D0 OBIS ASCII
; 9600@7E1 for OP-type devices, 9600@8N1 for SM-type devices
+1,3,o,0,9600,SM,1
; Zählerstand zu +A, tariflos, 
; Zählerstände Auflösung 10 µW*h (6 Vorkomma- und 8 Nachkommastellen)
1,1-0:1.8.0*255(@0.001,Energie Bezung,Wh,1_8_0,8
; Zählerstand zu +A, Tarif 1
1,1-0:1.8.1*255(@0.001,Energie Bezung T1,Wh,1_8_1,8
; Zählerstand zu +A, Tarif 2
1,1-0:1.8.2*255(@0.001,Energie Bezung T2,Wh,1_8_2,8
; Zählerstand zu -A, tariflos
1,1-0:2.8.0*255(@0.001,Energie Export,Wh,2_8_0,8
; Summe der Momentan-Leistungen in allen Phasen, Auflösung 0,01W (5 Vorkomma- und 2 Nachkommastellen)
1,1-0:16.7.0*255(@1,Leistung,W,16_7_0,18
; Momentane Leistung in Phase Lx, Auflösung 0,01W (5 Vorkomma- und 2 Nachkommastellen)
1,1-0:36.7.0*255(@1,Leistung L1,W,36_7_0,18
1,1-0:56.7.0*255(@1,Leistung L2,W,56_7_0,18
1,1-0:76.7.0*255(@1,Leistung L3,W,76_7_0,18
; Spannung in Phase Lx, Auflösung 0,1V (nur über MSB)
1,1-0:32.7.0*255(@1,Spannung L1,V,32_7_0,1
1,1-0:52.7.0*255(@1,Spannung L2,V,52_7_0,1
1,1-0:72.7.0*255(@1,Spannung L3,V,72_7_0,1
; Statuswort, 4 Byte Information über den Betriebszustand, HEX string
; tasmota can decode one string per device only!
;1,1-0:96.5.0*255(@#),Status1,,96_5_0,0
;1,1-0:96.8.0*255(@#),Status2,,96_8_0,0
; Geräte-Identifikation, Nach DIN 43863-5 
1,1-0:96.1.0*255(@#),Identifikation,,96_1_0,0
;1,1-0:0.0.0*255(@#),Identifikation,,0_0_0,0
#
`

Thanks in advance!

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: 12.0.2
  - [ ] Pre-compiled
  - [x] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by tis-cs** [comment, marked as answer]
I've to correct myself. The values of the counter are still available. They will be just refreshed if they are one full kWh.
After setting the pin in the smartmeter, tasmota is able to get more detailed consumption information like "SM Energie Bezug | 17739125.64471000 Wh" (8 decimals) which won't be published via mqtt like the others.

So, no issue and no discussion. Just a information :)

**Comment by ReneCode** [comment]
I have the same configuration ebz dd3 and tasmota.
via mqtt I get only tele/..../SENSOR  data with "Leistung (W)" - but I want also to get "Energie (kWh)"
how can I get this data ?

**Reply by tis-cs** [reply]
Already unlocked with the pin from your EVU?

**Reply by ReneCode** [reply]
yes I do.

In the meantime I got a message with all information I want to get (kWh, sum-W, L1-W, L2-W, L3-W)

` {"Time":"...","SM":{"1_8_0":126779.90357000,"16_7_0":117.10,"36_7_0":81.64,"56_7_0":27.88,"76_7_0":7.58}}`

And this message I get in that interval I can set in tasmota / Config / Logging / Telemetry period.   Beautiful.

but unfortunately I get all that "Leistung (W)" data in a separate message - and this is send *each second*

    20:33:09.086 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:09","SM":{"36_7_0":79.87}}
    20:33:09.162 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:09","SM":{"56_7_0":28.03}}
    20:33:09.237 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:09","SM":{"76_7_0":7.64}}
    20:33:10.013 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:09","SM":{"16_7_0":113.63}}
    20:33:10.093 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:10","SM":{"36_7_0":78.32}}
    20:33:10.167 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:10","SM":{"56_7_0":27.74}}
    20:33:10.241 MQT: tele/home/power/SENSOR = {"Time":"2022-11-03T20:33:10","SM":{"76_7_0":7.57}}

the topic of that single message is the same as the (prefered) combined message.

How can I get rid of that single messages ?

**Reply by matzelkoenig** [reply]
You can configure the behaviour of each meter metric via the **_\<precision\>_** parameter (see here: [Meter-Metrics](https://tasmota.github.io/docs/Smart-Meter-Interface/#meter-metrics))

This parameter determines how many decimal places the value should have. If you add up 16 here, then the measurement is immediately transmitted via MQTT with every change. Example: 2 = two decimal places and transmission set by TelePeriod. 18 = also 2 decimal places, but transmission with every change.

**Reply by Hardy3eich** [reply]
just to let you know: These eBz DD3 xxxx DTB - ODZ1 meter have two (2) IRs 
the one on the front is quite obvious and you need to put in a PIN to get all datas from there
and another on the top (maybe covered by a label as in picture below). This IR gives you even more datas (without PIN)
![zaehler](https://github.com/arendst/Tasmota/assets/133273946/952be8d5-acda-41fc-94ed-8d40c05d9405)

USB-port of Hitchi should face to front.

**Reply by sega890** [reply]
21:41:44.387 MQT: tele/tasmota_79822A/STATE = {"Time":"2023-07-20T21:41:44","Uptime":"0T00:11:20","UptimeSec":680,"Heap":19,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"MySSID ng","BSSId":"50:E6:36:5A:2D:8B","Channel":6,"Mode":"11n","RSSI":56,"Signal":-72,"LinkCount":1,"Downtime":"0T00:00:05"}}
21:41:44.398 MQT: tele/tasmota_79822A/SENSOR = {"Time":"2023-07-20T21:41:44","SM":{"1_8_0":7383.00,"2_8_0":933.00,"16_7_0":0.00,"36_7_0":0.00,"56_7_0":0.00,"76_7_0":0.00,"32_7_0":0.0,"52_7_0":0.0,"72_7_0":0.0,"96_1_0":"1EBZ0100817504"}}

Hi, just for clarification. I have the same eBZ device and got the PIN from my agency. Unlocking worked fine and I can see the current consuption and historic data in the display. Unfortunately, I only receive 0,00 values for SM Leistung 1-3.  A reboot does not help.

What can I do?

**Reply by sega890** [reply]
I figuered that out. My code works perfectly fine but I hat to move my Hichi IR sensor to the top of the eBZ smartmeter. There is another IR interface sending a complete dataset. It's solved now.
